### PR TITLE
Remove long-deprecated code

### PR DESCRIPTION
**Pull request by LeviPesin**
Related issue: -

**Description**

Remove some long-deprecated code and checks (hopefully all of them, although unsure).

### CODE REVIEW

**Comment by LeviPesin** [comment]
It had somehow broke something in https://github.com/mrdoob/three.js/pull/24431/commits/2b428c08f7b6c0fffbc9e5f5bdfe4811cbe7b663 and https://github.com/mrdoob/three.js/pull/24431/commits/9d1bac4236ff29fb17f23a27f6bf7fc3c435ccc0...

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -323,44 +316,16 @@ class ObjectLoader extends Loader {
...
-
-						array.push( cache[ material.uuid ] );
-
-					}
-
-					materials[ data.uuid ] = array;
-
-				} else {
-
-					if ( cache[ data.uuid ] === undefined ) {
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js:354]: Why did you remove this `if` statement?

Comment (LeviPesin) [reply src/loaders/ObjectLoader.js:354]: It is said there that it is deprecated. Or is it talking about the entire multi-materials usage?

Comment (Mugen87) [reply src/loaders/ObjectLoader.js:354]: Only the `MultiMaterial` branch is deprecated.

Comment (LeviPesin) [reply src/loaders/ObjectLoader.js:354]: I think I've removed only it? The other branch remains how it was.

File: src/loaders/ObjectLoader.js [hunk 2]
```diff
@@ -323,44 +316,16 @@ class ObjectLoader extends Loader {
...
-			for ( let i = 0, l = json.length; i < l; i ++ ) {
-
-				const data = json[ i ];
-
-				if ( data.type === 'MultiMaterial' ) {
-
-					// Deprecated
-
-					const array = [];
+			for ( const data of json ) {
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js:319]: Please do not change the structure of for loops.

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (Mugen87) [reply src/loaders/ObjectLoader.js:354]: Ah, the `if ( cache[ data.uuid ] === undefined ) {` is further above 👍 .

File: src/loaders/ObjectLoader.js (hunk 2 above)
Comment (LeviPesin) [reply src/loaders/ObjectLoader.js:319]: Not sure, why (I think for-of loops are much simpler and easier to read), but OK, reverted.

**Comment by Mugen87** [comment]
It seems the PR still breaks some examples...

**Comment by LeviPesin** [comment]
Should be fixed now.

File: examples/jsm/loaders/GLTFLoader.js [hunk 3]
```diff
@@ -1487,7 +1487,7 @@ class GLTFDracoMeshCompressionExtension {
 				const accessorDef = json.accessors[ primitive.attributes[ attributeName ] ];
 				const componentType = WEBGL_COMPONENT_TYPES[ accessorDef.componentType ];
 
-				attributeTypeMap[ threeAttributeName ] = componentType;
+				attributeTypeMap[ threeAttributeName ] = componentType.name;
```
Comment (Mugen87) [inline examples/jsm/loaders/GLTFLoader.js:1490]: I'm confused. Why is this change now necessary?

Comment (LeviPesin) [reply examples/jsm/loaders/GLTFLoader.js:1490]: Because I have removed the code from the DRACOLoader which auto-converts types in `attributeTypeMap` from constructors like `Float32Array` to strings like `'Float32Array'`.

**Review by Mugen87 — APPROVED** [review]

File: src/loaders/ObjectLoader.js (hunk 2 above)
Comment (gkjohnson) [reply src/loaders/ObjectLoader.js]: > Not sure, why

Performance

**Comment by mrdoob** [comment]
Thanks!
